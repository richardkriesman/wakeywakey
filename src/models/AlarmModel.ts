export enum DayOfWeek { Sunday, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday }

export interface AlarmModel {
    days: DayOfWeek[];
    sleepTime: Date;
    wakeUpTime: Date;
    getUpTime: Date;
}

export const TestAlarms: AlarmModel[] = [
    {
        days: [ DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday ],
        getUpTime: new Date(),
        sleepTime: new Date(),
        wakeUpTime: new Date()
    },
    {
        days: [ DayOfWeek.Sunday, DayOfWeek.Saturday ],
        getUpTime: new Date(),
        sleepTime: new Date(),
        wakeUpTime: new Date()
    }
];
