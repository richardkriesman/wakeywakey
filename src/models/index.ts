export { Alarm } from "./Alarm";
export { Schedule } from "./Schedule";
