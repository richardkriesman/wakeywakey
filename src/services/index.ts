export { PasscodeService } from "./PasscodeService";
export { ScheduleService } from "./ScheduleService";
