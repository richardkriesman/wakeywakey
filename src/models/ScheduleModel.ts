import { AlarmModel } from "./AlarmModel";

export interface ScheduleModel {
    alarms: AlarmModel[];
    enabled: boolean;
    key: number;
    name: string;
}
