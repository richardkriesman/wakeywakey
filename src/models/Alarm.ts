import { AppDatabase } from "../utils/AppDatabase";
import { Model } from "../utils/Model";
import { Schedule } from "./Schedule";
import { Time } from "../utils/Time";

export enum AlarmDay {
    Monday = 0,
    Tuesday = 1,
    Wednesday = 2,
    Thursday = 3,
    Friday = 4,
    Saturday = 5,
    Sunday = 6
}

/**
 * An Alarm specifies the times go to bed or get up. It can be repeated over a specified number of days.
 *
 * Alarms have 3 event times:
 *  - Sleep:   The time the child should to go to sleep
 *  - Wake up: The time the child should wake up
 *  - Get up:  The time the child is allowed to get up
 *
 *  When an event time is reached and the Alarm's parent Schedule is enabled, the alarm sound as specified in the
 *  Schedule will be played.
 *
 *  A Schedule can have multiple alarms, but only such that no two Alarms are assigned to any day.
 */
export class Alarm extends Model {

    /**
     * Creates a new Alarm.
     *
     * For internal use only. Use a {@link Schedule} to create a new alarm from the UI.
     *
     * @param db Database connection
     * @param scheduleId ID of the Schedule to which this alarm is attached
     * @param sleepTime Time the child should go to sleep
     * @param wakeTime Time the child should wake up
     * @param getUpTime Time the child is allowed to get up
     * @param days Days of the week the Alarm is active for
     */
    public static async create(db: AppDatabase, scheduleId: number, sleepTime: Time, wakeTime: Time,
                               getUpTime: Time, days: AlarmDay[]): Promise<Alarm> {

        // insert the alarm into the database
        const result: SQLResultSet = await db.execute(`
            INSERT INTO alarm
                (scheduleId, sleepTime, wakeTime, getUpTime)
            VALUES
                (?, ?, ?, ?)
        `, [scheduleId, sleepTime.totalSeconds, wakeTime.totalSeconds, getUpTime.totalSeconds]);

        // TODO: Add days of week

        // update the watchers
        db.getEmitterSet<Alarm>(Alarm.name).update(await Alarm.getAll(db));

        // build the resulting model
        return Alarm.load(db, {
            getUpTime: getUpTime.totalSeconds,
            id: result.insertId,
            scheduleId,
            sleepTime: sleepTime.totalSeconds,
            wakeTime: wakeTime.totalSeconds
        });
    }

    /**
     * Gets all {@link Alarm}s.
     *
     * For internal use only.
     *
     * @param db Database connection
     */
    public static async getAll(db: AppDatabase): Promise<Alarm[]> {
        const result: SQLResultSet = await db.execute(`
            SELECT *
            FROM alarm;
        `);
        const alarms: Alarm[] = [];
        for (let i = 0; i < result.rows.length; i++) {
            alarms.push(Alarm.load(db, result.rows.item(i)));
        }
        return alarms;
    }

    /**
     * Gets an {@link Alarm} by its ID.
     *
     * For internal use only.
     *
     * @param db Database connection
     * @param id ID of the Alarm to retrieve
     */
    public static async getById(db: AppDatabase, id: number): Promise<Alarm|undefined> {
        const result: SQLResultSet = await db.execute(`
            SELECT *
            FROM alarm
            WHERE
                id = ?
        `, [id]);
        return result.rows.length > 0 ? Alarm.load(db, result.rows.item(0)) : undefined;
    }

    /**
     * Gets an {@link Alarm} by its {@link Schedule}'s ID.
     *
     * For internal use only.
     *
     * @param db Database connection
     * @param scheduleId ID of the Schedule to which the Alarms are attached
     */
    public static async getByScheduleId(db: AppDatabase, scheduleId: number): Promise<Alarm[]> {
        const result: SQLResultSet = await db.execute(`
            SELECT *
            FROM alarm
            WHERE
                scheduleId = ?
        `, [scheduleId]);
        const alarms: Alarm[] = [];
        for (let i = 0; i < result.rows.length; i++) {
            alarms.push(Alarm.load(db, result.rows.item(i)));
        }
        return alarms;
    }

    /**
     * Generates a model given a row from the database.
     *
     * @param db Database connection
     * @param row Row from the database
     */
    private static load(db: AppDatabase, row: any): Alarm {
        const model = new Alarm(db, row.scheduleId);
        model._id = row.id;
        model._sleepTime = Time.createFromTotalSeconds(row.sleepTime);
        model._wakeTime = Time.createFromTotalSeconds(row.wakeTime);
        model._getUpTime = Time.createFromTotalSeconds(row.getUpTime);
        return model;
    }

    /**
     * ID of the {@link Schedule} to which this Alarm is attached.
     */
    public readonly scheduleId: number;

    private _sleepTime: Time;
    private _wakeTime: Time;
    private _getUpTime: Time;

    private constructor(db: AppDatabase, scheduleId: number) {
        super(db);
        this.scheduleId = scheduleId;
    }

    /**
     * Time the child should go to sleep as the number of seconds from the start of the day.
     */
    public get sleepTime(): Time {
        return this._sleepTime;
    }

    /**
     * Time the child should wake up as the number of seconds from the start of the day.
     */
    public get wakeTime(): Time {
        return this._wakeTime
    }

    /**
     * Time the child is allowed to get up as the number of seconds from the start of the day.
     */
    public get getUpTime(): Time {
        return this._getUpTime;
    }

    /**
     * Deletes the Alarm.
     *
     * Do not use this Alarm after it has been deleted.
     */
    public async delete(): Promise<void> {
        await this.db.execute(`
            DELETE FROM alarm
            WHERE
                id = ?
        `, [this.id]);
        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await Alarm.getAll(this.db));
    }

    public async setSleepTime(time: Time): Promise<void> {
        await this.db.execute(`
            UPDATE alarm
            SET
                sleepTime = ?
        `, [time.totalSeconds]);
        this._sleepTime = time;
        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await Alarm.getAll(this.db));
    }

    public async setWakeTime(time: Time): Promise<void> {
        await this.db.execute(`
            UPDATE alarm
            SET
                wakeTime = ?
        `, [time.totalSeconds]);
        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await Alarm.getAll(this.db));
    }

    public async setGetUpTime(time: Time): Promise<void> {
        await this.db.execute(`
            UPDATE alarm
            SET
                getUpTime = ?
        `, [time.totalSeconds]);
        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await Alarm.getAll(this.db));
    }

}
