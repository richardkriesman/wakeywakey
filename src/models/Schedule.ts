import { AppDatabase } from "../utils/AppDatabase";
import { Model } from "../utils/Model";
import { Alarm } from "./Alarm";
import { Watcher } from "../utils/watcher/Watcher";
import { Emitter } from "../utils/watcher/Emitter";
import {Time} from "../utils/Time";

/**
 * A Schedule represents a weekly alarm schedule, such as "School Nights" for weeks when a child has to get up earlier
 * than normal to go to school.
 *
 * A Schedule contains alarms, with no more than 1 alarm assigned to each day of the week. It also contains preferences
 * specific to that schedule, such as alarm sounds and snooze limits.
 */
export class Schedule extends Model {

    public static readonly NAME_MAX_LENGTH: number = 30;

    /**
     * Creates a new Schedule.
     *
     * For internal use only. Use {@link ScheduleService} to create a new schedule from the UI.
     *
     * @param db Database connection
     * @param name Non-unique name for the schedule
     */
    public static async create(db: AppDatabase, name: string): Promise<Schedule> {

        // insert the schedule into the database
        const result: SQLResultSet = await db.execute(`
            INSERT INTO schedule
                (name, isEnabled)
            VALUES
                (?, 0)
        `, [name]);

        // build the resulting model
        return Schedule.load(db, {
            id: result.insertId,
            isEnabled: false,
            name
        });
    }

    /**
     * Deletes a Schedule.
     *
     * For internal use only. Use {@link ScheduleService} to delete a schedule from the UI.
     *
     * @param db Database connection
     * @param schedule Schedule to be deleted
     */
    public static async delete(db: AppDatabase, schedule: Schedule): Promise<void> {
        await db.execute(`
            DELETE FROM schedule
            WHERE
                id = ?
        `, [schedule.id]);
    }

    /**
     * Gets all Schedules.
     *
     * For internal use only. Use {@link ScheduleService} to get all Schedules from the UI.
     *
     * @param db Database connection
     */
    public static async getAll(db: AppDatabase): Promise<Schedule[]> {

        // get all rows in ascending order
        const result: SQLResultSet = await db.execute(`
            SELECT *
            FROM schedule;     
        `);

        // build models from results
        const models: Schedule[] = [];
        for (let i = 0; i < result.rows.length; i++) {
            models.push(Schedule.load(db, result.rows.item(i)));
        }
        return models;
    }

    /**
     * Generates a model given a row from the database.
     *
     * @param db Database connection
     * @param row Row from the database
     */
    private static load(db: AppDatabase, row: any): Schedule {
        const model = new Schedule(db);
        model._id = row.id;
        model._name = row.name;
        model._isEnabled = !!row.isEnabled;
        return model;
    }

    private _name: string;
    private _isEnabled: boolean;

    /**
     * A non-unique, user-friendly name
     */
    public get name(): string {
        return this._name;
    }

    /**
     * Whether the Schedule is enabled. Only one Schedule can be enabled at a time.
     */
    public get isEnabled(): boolean {
        return this._isEnabled;
    }

    /**
     * Creates a new {@link Alarm} attached to this Schedule.
     *
     * @param sleepTime Time the child should go to sleep
     * @param wakeTime Time the child should wake up
     * @param getUpTime Time the child is allowed to get up
     */
    public createAlarm(sleepTime: Time, wakeTime: Time, getUpTime: Time): Promise<Alarm> {
        return Alarm.create(this.db, this.id, sleepTime, wakeTime, getUpTime, []);
    }

    /**
     * Returns a {@link Watcher} that updates with this {@link Schedule}'s set of {@link Alarm}s.
     */
    public watchAlarms(): Watcher<Alarm> {
        const emitter: Emitter<Alarm> = this.db.getEmitterSet<Alarm>(Alarm.name).create();

        // configure filter
        emitter.onFilter((alarms: Alarm[]) => {
            const newAlarms: Alarm[] = [];
            for (const alarm of alarms) {
                if (alarm.scheduleId === this.id) {
                    newAlarms.push(alarm);
                }
            }
            return newAlarms;
        });

        // get initial data set
        Alarm.getByScheduleId(this.db, this.id)
            .then((alarms: Alarm[]) => {
                emitter.updateInitialSet(alarms);
            });

        return emitter;
    }

}
