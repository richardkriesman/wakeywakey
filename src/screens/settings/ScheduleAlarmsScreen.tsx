/**
 * @module screens
 */

import React, { ReactNode } from "react";
import { FlatList, StyleSheet, View } from "react-native";
import { ListItem, Text } from "react-native-elements";
import { NavigationScreenProps, StackActions } from "react-navigation";

import Colors from "../../constants/Colors";
import Layout from "../../constants/Layout";
import { AlarmModel } from "../../models/AlarmModel";
import { ScheduleModel } from "../../models/ScheduleModel";
import { BottomTabBarIcon, Title } from "../../utils/screen/NavigationOptions";
import { UIScreen } from "../../utils/screen/UIScreen";

export interface EditScheduleScreenState {
    headerTitle: string;
    schedule?: ScheduleModel;
}

@BottomTabBarIcon("ios-alarm") @Title("Alarms")
export default class ScheduleAlarmsScreen extends UIScreen<{}, EditScheduleScreenState> {

    private static padTime(time: number): string {
        return ("0" + time).slice(-2);
    }

    private static formatTime(date: Date): string {
        return `${this.padTime(date.getHours())}:${this.padTime(date.getMinutes())}`;
    }

    // noinspection JSUnusedLocalSymbols
    private static getAlarmTitle(alarm: AlarmModel): string {
        const start = this.formatTime(alarm.sleepTime);
        const end = this.formatTime(alarm.getUpTime);
        return `${start} â€“ ${end}`;
    }

    // noinspection JSUnusedLocalSymbols
    private static getAlarmSubtitle(alarm: AlarmModel): string {
        return alarm.days.join(", ");
    }

    public constructor(props: NavigationScreenProps) {
        super(props);
    }

    public componentWillMount(): void {
        this.setState({
            headerTitle: this.props.navigation.getParam("title"),
            schedule: this.props.navigation.getParam("schedule") || null
        });
    }

    public onAlarmPressed(key: number): void {
        this.props.navigation.dispatch(StackActions.push({
            params: {
                alarm: this.state.schedule.alarms[key],
                title: this.state.headerTitle
            },
            routeName: "EditAlarm"
        }));
    }

    public renderContent(): ReactNode {
        return (
            <View style={styles.viewScroller}>
                <Text style={styles.textSectionHeader}>Alarms</Text>

                <FlatList
                    data={this.state.schedule ? this.state.schedule.alarms : []}
                    keyExtractor={(item: AlarmModel): string => String(item.key)}
                    renderItem={({ item }) => (
                        <ListItem
                            onPress={this.onAlarmPressed.bind(this, item.key)}
                            title={ScheduleAlarmsScreen.getAlarmTitle(item)}
                            subtitle={ScheduleAlarmsScreen.getAlarmSubtitle(item)}
                            rightIcon={{ name: "arrow-forward", type: "ionicons" }}
                        />
                    )}
                />

            </View>
        );
    }
}

const styles = StyleSheet.create({
    daySelector: {
        flexDirection: "row",
        justifyContent: "space-around"
    },
    deleteButton: {
        backgroundColor: Colors.appleButtonRed
    },
    deleteButtonContainer: {
        padding: 20
    },
    deleteButtonTitle: {
        color: "#fff"
    },
    divider: {
        backgroundColor: Colors.headerBackground,
        marginBottom: 20,
        marginTop: 10
    },
    saveButton: {
        color: Colors.appleButtonBlue,
        fontWeight: "500",
        marginRight: 10
    },
    textSectionHeader: {
        color: Colors.subheaderColor,
        fontSize: 17,
        fontWeight: "600",
        marginBottom: 10
    },
    viewScroller: {
        height: Layout.window.height,
        padding: 20
    }
});
