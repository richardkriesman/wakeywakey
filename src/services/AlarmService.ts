/**
 * @module services
 */

import { Alarm } from "../models/Alarm";
import { Schedule } from "../models/Schedule";
import { Service } from "../utils/Service";
import { Time } from "../utils/Time";
import { Emitter } from "../utils/watcher/Emitter";
import { Watcher } from "../utils/watcher/Watcher";

export class AlarmService extends Service {

    /**
     * Creates a new Alarm.
     *
     * Days are selected by using the bitwise OR operation like so:
     *     `AlarmDays.Monday | AlarmDays.Tuesday`
     *
     * @param schedule Schedule to which this alarm is attached
     * @param sleepTime Time the child should go to sleep
     * @param wakeTime Time the child should wake up
     * @param getUpTime Time the child is allowed to get up
     * @param days Days of the week the Alarm is active for as an integer
     */
    public async create(schedule: Schedule, sleepTime: Time, wakeTime: Time, getUpTime: Time,
                        days: number): Promise<Alarm> {

        // insert the alarm into the database
        const result: SQLResultSet = await this.db.execute(`
            INSERT INTO alarm
                (scheduleId, days, sleepTime, wakeTime, getUpTime)
            VALUES
                (?, ?, ?, ?, ?)
        `, [schedule.id, days, sleepTime.totalSeconds, wakeTime.totalSeconds, getUpTime.totalSeconds]);

        // update the emitters
        this.db.getEmitterSet<Alarm>(Alarm.name).update(await this.getAll());

        // build the resulting model
        return Alarm.load(this.db, {
            getUpTime: getUpTime.totalSeconds,
            id: result.insertId,
            schedule,
            sleepTime: sleepTime.totalSeconds,
            wakeTime: wakeTime.totalSeconds
        });

    }

    /**
     * Deletes the Alarm.
     *
     * Do not use the Alarm after it has been deleted.
     */
    public async delete(alarm: Alarm): Promise<void> {
        await this.db.execute(`
            DELETE FROM alarm
            WHERE
                id = ?
        `, [alarm.id]);
        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await this.getAll());
    }

    /**
     * Gets an {@link Alarm} by its ID.
     *
     * @param id ID of the Alarm to retrieve
     */
    public async get(id: number): Promise<Alarm | undefined> {
        const result: SQLResultSet = await this.db.execute(`
            SELECT *
            FROM alarm
            WHERE
                id = ?
        `, [id]);
        return result.rows.length > 0 ? Alarm.load(this.db, result.rows.item(0)) : undefined;
    }

    /**
     * Gets all {@link Alarm}s for all {@link Schedule}s.
     */
    public async getAll(): Promise<Alarm[]> {
        const result: SQLResultSet = await this.db.execute(`
            SELECT *
            FROM alarm;
        `);
        const alarms: Alarm[] = [];
        for (let i = 0; i < result.rows.length; i++) {
            alarms.push(Alarm.load(this.db, result.rows.item(i)));
        }
        return alarms;
    }

    /**
     * Gets {@link Alarm}s for a {@link Schedule}.
     *
     * @param schedule Schedule to which the Alarms are attached
     */
    public async getBySchedule(schedule: Schedule): Promise<Alarm[]> {
        const result: SQLResultSet = await this.db.execute(`
            SELECT *
            FROM alarm
            WHERE
                scheduleId = ?
        `, [schedule.id]);
        const alarms: Alarm[] = [];
        for (let i = 0; i < result.rows.length; i++) {
            alarms.push(Alarm.load(this.db, result.rows.item(i)));
        }
        return alarms;
    }

    /**
     * Updates the days the {@link Alarm} is active.
     *
     * @param alarm Alarm to update
     * @param days New days the Alarm should be active
     */
    public async setDays(alarm: Alarm, days: number): Promise<Alarm> {
        await this.db.execute(`
            UPDATE alarm
            SET
                days = ?
            WHERE
                id = ?
        `, [days, alarm.id]);

        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await this.getAll());

        const data: any = Alarm.save(alarm);
        data.days = days;
        return Alarm.load(this.db, data);
    }

    /**
     * Updates the get up time for the {@link Alarm}.
     *
     * @param alarm Alarm to update
     * @param time New get up time
     */
    public async setGetUpTime(alarm: Alarm, time: Time): Promise<Alarm> {
        await this.db.execute(`
            UPDATE alarm
            SET
                getUpTime = ?
            WHERE
                id = ?
        `, [time.totalSeconds, alarm.id]);

        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await this.getAll());

        const data: any = Alarm.save(alarm);
        data.getUpTime = time.totalSeconds;
        return Alarm.load(this.db, data);
    }

    /**
     * Updates the sleep time for the {@link Alarm}.
     *
     * @param alarm Alarm to update
     * @param time New sleep time
     */
    public async setSleepTime(alarm: Alarm, time: Time): Promise<Alarm> {
        await this.db.execute(`
            UPDATE alarm
            SET
                sleepTime = ?
            WHERE
                id = ?
        `, [time.totalSeconds, alarm.id]);

        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await this.getAll());

        const data: any = Alarm.save(alarm);
        data.sleepTime = time.totalSeconds;
        return Alarm.load(this.db, data);
    }

    /**
     * Updates the wake time for the {@link Alarm}.
     *
     * @param alarm Alarm to update
     * @param time New wake time
     */
    public async setWakeTime(alarm: Alarm, time: Time): Promise<Alarm> {
        await this.db.execute(`
            UPDATE alarm
            SET
                wakeTime = ?
            WHERE
                id = ?
        `, [time.totalSeconds, alarm.id]);

        await this.db.getEmitterSet<Alarm>(Alarm.name).update(await this.getAll());

        const data: any = Alarm.save(alarm);
        data.wakeTime = time.totalSeconds;
        return Alarm.load(this.db, data);
    }

    /**
     * Watch all {@link Alarm}s attached to the given {@link Schedule}.
     *
     * @param schedule Schedule whose Alarms to watch
     */
    public watchBySchedule(schedule: Schedule): Watcher<Alarm> {
        const emitter: Emitter<Alarm> = this.db.getEmitterSet<Alarm>(Alarm.name).create();

        // configure filter
        emitter.onFilter((alarms: Alarm[]) => { // filter out alarms not attached to this schedule
            const newAlarms: Alarm[] = [];
            for (const alarm of alarms) {
                if (alarm.scheduleId === schedule.id) {
                    newAlarms.push(alarm);
                }
            }
            return newAlarms;
        });

        // get initial data set
        this.getBySchedule(schedule)
            .then((alarms: Alarm[]) => {
                emitter.updateInitialSet(alarms);
            });

        return emitter;
    }

}
