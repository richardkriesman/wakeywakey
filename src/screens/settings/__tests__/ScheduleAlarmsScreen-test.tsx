import * as React from "react";
import renderer from "react-test-renderer";
import { Schedule } from "../../../models";
import { ScheduleService } from "../../../services";
import { TestEnvironment } from "../../../utils/testing";
import ScheduleAlarmsScreen from "../AlarmListScreen";

jest.mock("../../../services/ScheduleService");

let env: TestEnvironment;
let schedule: Schedule;
beforeEach((done) => {
    TestEnvironment.init()
        .then((newEnv) => {
            env = newEnv;

            // create a new schedule
            return env.db.getService(ScheduleService).create("Test schedule");
        })
        .then((newSchedule) => {
            schedule = newSchedule;
            done();
        });
});

it("renders correctly", () => {

    // build props
    const props = env.emptyUIScreenProps;
    props.navigation.setParams({ schedule });

    // render tree
    const tree = renderer.create(<ScheduleAlarmsScreen {...props}/>);
    expect(tree.toJSON()).toMatchSnapshot();

});
