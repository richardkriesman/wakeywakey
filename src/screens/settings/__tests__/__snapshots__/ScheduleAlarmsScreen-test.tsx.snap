// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      No alarms yet
    </Text>
    <Text
      style={
        Object {
          "color": "#5e5e5e",
          "fontSize": 15,
        }
      }
    >
      Create an alarm to set your child's bedtime
    </Text>
  </View>
</RCTSafeAreaView>
`;
